### PR TITLE
Add mutations to Recipes GraphQL server

### PR DESCRIPTION
**Pull request by mxstbr**
This adds resource mutations to the Recipes GraphQL server. For every resource, we generate `createX(x: XInput)`, `updateX(x: XInput)` and `destroyX(x: XInput)` mutations. This is what that looks like right now:

Screenshot 2020-04-21 at 10 31 53

### CODE REVIEW

File: packages/gatsby-recipes/src/create-types.js [hunk 1]
```diff
@@ -62,20 +66,84 @@ module.exports = () => {
...
+  const mutationTypes = _.flatten(
+    resourceTypes.filter(Boolean).map(r => r.mutation)
+  ).reduce((acc, curr) => {
+    Object.keys(curr).forEach(key => {
+      acc[typeNameToHumanName(key)] = curr[key]
+    })
+    return acc
+  }, {})
+
+  console.log(mutationTypes)
```
Comment (LekoArts) [inline packages/gatsby-recipes/src/create-types.js:143]: Stray `console.log`

**Comment by LekoArts** [comment]
Would it be sensible to update the current test for this file or add more tests? The current test seems a bit sparse.

File: packages/gatsby-recipes/src/create-types.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ const Joi2GQL = require(`./joi-to-graphql`)
 const Joi = require(`@hapi/joi`)
 const { GraphQLString, GraphQLObjectType, GraphQLList } = require(`graphql`)
 const _ = require(`lodash`)
+const { ObjectTypeComposer, schemaComposer } = require(`graphql-compose`)
```
Comment (KyleAMathews) [inline packages/gatsby-recipes/src/create-types.js:5]: this is a convenient helper function!

Comment (mxstbr) [reply packages/gatsby-recipes/src/create-types.js:5]: Super nice! Glad I stumbled upon it, shoutout to @freiksent who pointed it out in some really old graphql.js issue :fire:

**Comment by mxstbr** [comment]
Arg because I had to squash and merge the merge commit into this branch the diff is now totally messed up :confused:

**Comment by mxstbr** [comment]
Merged into #23291 manually to avoid the squash and merge!

**Pull request closed without merging**